### PR TITLE
Node 267 sync fork application 2

### PR DESCRIPTION
**Pull request by ismagin**
- De-synchronized UTX pool
- Introduced peaceful(`Right(None)`) declining of block if better version exists 
- `CoordinatorHandler` via monix executor to reduce entropy
- Check nodes are not deadlocked before stop

### CODE REVIEW

**Comment by ismagin** [comment]
I don't like the idea to lock utx and state with one RWL, but currently it seems to be the easiest fix(contrary to rewriting UTX w/o lock)

**Comment by ismagin** [comment]
I also had to introduce silent non-application of a block similar, which is not necessary, but it came up with re-application of unsynced version of fork.

File: src/test/scala/com/wavesplatform/UtxPoolSpecification.scala [hunk 1]
```diff
@@ -48,6 +48,8 @@ class UtxPoolSpecification extends FreeSpec
     (state, history)
   }
 
+  def lock = new ReentrantReadWriteLock(true)
```
Comment (vsuharnikov) [inline src/test/scala/com/wavesplatform/UtxPoolSpecification.scala]: Looks like a non-used code :)

File: src/main/scala/scorex/utils/Synchronized.scala [hunk 2]
```diff
@@ -89,7 +89,7 @@ trait Synchronized extends ScorexLogging {
     }
     finally {
       lock.unlock()
-//      log.trace(s"unlocked $lock")
+      //      log.trace(s"unlocked $lock")
```
Comment (alexeykiselev) [inline src/main/scala/scorex/utils/Synchronized.scala]: maybe better remove

**Pull request merged**